drivers/serial/serial.c

```diff
@@ -1771,7 +1771,18 @@ void uart_connected(FAR uart_dev_t *dev, bool connected)
 void uart_reset_sem(FAR uart_dev_t *dev)
 {
   nxsem_reset(&dev->xmitsem,  0);
-  nxsem_reset(&dev->recvsem,  1);
+
+  /* Wake up reading function if it is blocked */
+
+  if (dev->recvwaiting)
```

Comment:
What about the recvwaiting life cycle (being reset) and any pollwaiters? What releases them? The LL driver would call "uart_datareceived"

Reply:
I was sure polls are canceled at higher level of nx_close. But they are not.

We can add
**uart_pollnotify(dev, POLLIN);**
to the close function.

This is not the logic any low level driver should care of.

Reply:
Can you just use uart_datareceived to release  the waiters and the rx thread?

Reply:
OK!